### PR TITLE
Remove client-side chat prediction.

### PR DESCRIPTION
**Pull request by red-001**
Network lag isn't really a big issue with chat and chat prediction makes writing mods harder. See #4976.

### CODE REVIEW

File: src/server.cpp [hunk 1]
```diff
@@ -2832,6 +2832,9 @@ std::wstring Server::handleChat(const std::string &name, const std::wstring &wna
 			if (cid != peer_id_to_avoid_sending)
 				SendChatMessage(cid, line);
```
Comment (rubenwardy) [inline src/server.cpp]: don't you need to remove this if statement?

File: src/server.cpp [hunk 2]
```diff
@@ -2832,6 +2832,9 @@ std::wstring Server::handleChat(const std::string &name, const std::wstring &wna
...
 		}
+		if (player->protocol_version >= 30){
```
Comment (nerzhul) [inline src/server.cpp]: wrong return, you should only remove filter in the upper side

File: src/client.cpp [hunk 3]
```diff
@@ -1563,10 +1563,12 @@ void Client::typeChatMessage(const std::wstring &message)
 	}
 	else
 	{
+		if (m_proto_ver < 30){
 		LocalPlayer *player = m_env.getLocalPlayer();
 		assert(player != NULL);
 		std::wstring name = narrow_to_wide(player->getName());
 		m_chat_queue.push((std::wstring)L"<" + name + L"> " + message);
+		}
```
Comment (sfan5) [inline src/client.cpp]: please indent the lines

File: src/client.cpp [hunk 4]
```diff
@@ -1563,10 +1563,12 @@ void Client::typeChatMessage(const std::wstring &message)
 	}
 	else
 	{
+		if (m_proto_ver < 30){
```
Comment (sfan5) [inline src/client.cpp]: spacing between `)` and `{`

File: src/network/networkprotocol.h [hunk 5]
```diff
@@ -142,9 +142,11 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 		Server doesn't accept TOSERVER_BREATH anymore
 		serialization of TileAnimation params changed
 		TAT_SHEET_2D
+	PROTOCOL_VERSION 30:
+		Removed client-sided chat perdiction
 */
 
-#define LATEST_PROTOCOL_VERSION 29
+#define LATEST_PROTOCOL_VERSION 30
```
Comment (sfan5) [inline src/network/networkprotocol.h]: don't bump yet again, keep it at 29

File: src/server.cpp (hunk 2 above)
Comment (sfan5) [inline src/server.cpp]: again spacing between `)` and `{`

Comment (red-001) [inline src/server.cpp]: ok will change that

File: src/network/networkprotocol.h (hunk 5 above)
Comment (red-001) [inline src/network/networkprotocol.h]: ok

**Comment by nerzhul** [comment]
Seems good to me

File: src/server.cpp [hunk 6]
```diff
@@ -2826,7 +2826,15 @@ std::wstring Server::handleChat(const std::string &name, const std::wstring &wna
 
 		std::vector<u16> clients = m_clients.getClientIDs();
 
+		/*
+			Don't send the message back to the inital sender
+			if they are using protocol version < 29
```
Comment (nerzhul) [inline src/server.cpp]: typo to fix, version > 29

**Pull request merged**